openmetadata-service/src/main/java/org/openmetadata/service/apps/ExternalApplicationHandler.java

```diff
@@ -0,0 +1,111 @@
+package org.openmetadata.service.apps;
+
+import com.cronutils.model.Cron;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.schema.api.services.ingestionPipelines.CreateIngestionPipeline;
+import org.openmetadata.schema.entity.applications.configuration.ExternalApplicationConfig;
+import org.openmetadata.schema.entity.services.ingestionPipelines.AirflowConfig;
+import org.openmetadata.schema.entity.services.ingestionPipelines.IngestionPipeline;
+import org.openmetadata.schema.entity.services.ingestionPipelines.PipelineType;
+import org.openmetadata.schema.metadataIngestion.ApplicationPipeline;
+import org.openmetadata.schema.metadataIngestion.SourceConfig;
+import org.openmetadata.schema.type.EntityReference;
+import org.openmetadata.schema.type.ProviderType;
+import org.openmetadata.schema.type.Relationship;
+import org.openmetadata.service.Entity;
+import org.openmetadata.service.exception.EntityNotFoundException;
+import org.openmetadata.service.jdbi3.CollectionDAO;
+import org.openmetadata.service.jdbi3.IngestionPipelineRepository;
+import org.openmetadata.service.jdbi3.MetadataServiceRepository;
+import org.openmetadata.service.util.FullyQualifiedName;
+import org.openmetadata.service.util.JsonUtils;
+
+@Slf4j
+public class ExternalApplicationHandler extends AbstractNativeApplication {
+  private static final String SERVICE_NAME = "OpenMetadata";
+
+  /**
+   * MetaPilot is an external App that accepts one ApiKey parameter and runs a workflow based on it.
+   *
+   * <p>The App will register an IngestionPipeline against the OpenMetadata service with default daily scheduled
+   * configurations.
+   */
+  @Override
+  public void initializeExternalApp() {
```

Comment:
is this one-time call? we need to have another method where the application config can be called as well, during the installation and also if the user makes any changes

Reply:
yes this is one time. I will be adding the configure method we discussed in a followup patch